### DISCUSSION TITLE
Welcome to hands-on-llms Discussions!

### DISCUSSION THREAD

**Discussion by iusztinpaul** [iusztinpaul/hands-on-llms#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect to share the progress of the course and connect with the people who want to try it out. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Follow along with our course implementation.

**Comment by MaruthiKo** [comment]
Hi Paul

Can we learn LangChain without using the OpenAI API for free of cost?

I have tried several ways of downloading using open-source LLM models but as I'm not having any GPUs they work very very slow.
I am a student and it's hard to afford a GPU.
Any suggestions on how to learn langchain for free of cost?

**Reply by iusztinpaul** [reply]
Hello,

Try to use a tiny model that runs fast on your CPU. 

The results won't be great, but you can play with the Langchain ecosystem for free.

**Reply by MaruthiKo** [reply]
okay thank you :)

**Reply by gregorywmorris** [reply]
@MaruthiKo  you can access a free GPU with Google colab. It is limited in that it may time out on you for long-running processes. But for learning it's great. 

There is also a way to run a script in the browser that may prevent it from timing out if inclined to research that. It simply selects something on the page every 30 seconds.
